### PR TITLE
Various improvements and fixes

### PR DESCRIPTION
**Pull request by hanzei**
Please take a look a the individual commits.

Follow up on https://github.com/mattermost/mattermost-plugin-alertmanager/pull/1#pullrequestreview-1189661936

### CODE REVIEW

**Comment by hanzei** [comment]
cc @javaguirre

**Review by javaguirre — APPROVED** [review]
Thank you Ben! LGTM!

File: server/alertmanager/retry.go [hunk 1]
```diff
@@ -39,11 +39,11 @@ func httpRetry(method string, url string) (*http.Response, error) {
 		switch method {
 		case http.MethodGet:
 			if resp.StatusCode != http.StatusOK {
-				return fmt.Errorf("status code is %d not 200", resp.StatusCode)
+				return errors.Errorf("status code is %d not 200", resp.StatusCode)
```
Comment (cpanato) [inline server/alertmanager/retry.go:42]: github.com/pkg/errors is deprecated, please keep using `fmt.errorf`

File: server/alertmanager/retry.go [hunk 2]
```diff
@@ -39,11 +39,11 @@ func httpRetry(method string, url string) (*http.Response, error) {
...
 			}
 		case http.MethodPost:
 			if resp.StatusCode == http.StatusBadRequest {
-				return fmt.Errorf("status code is %d not 3xx", resp.StatusCode)
+				return errors.Errorf("status code is %d not 3xx", resp.StatusCode)
```
Comment (cpanato) [inline server/alertmanager/retry.go:46]: same here

File: server/alertmanager/silences.go [hunk 3]
```diff
@@ -36,7 +37,7 @@ func ListSilences(alertmanagerURL string) ([]types.Silence, error) {
 // DeleteSilence delete a silence by ID.
 func ExpireSilence(silenceID, alertmanagerURL string) error {
 	if silenceID == "" {
-		return fmt.Errorf("silence ID cannot be empty")
+		return errors.Errorf("silence ID cannot be empty")
```
Comment (cpanato) [inline server/alertmanager/silences.go:40]: github.com/pkg/errors is deprecated, please keep using `fmt.errorf`

File: server/alertmanager/silences.go [hunk 4]
```diff
@@ -51,7 +52,7 @@ func ExpireSilence(silenceID, alertmanagerURL string) error {
 		return err
 	}
 	if resp.StatusCode != 200 {
-		return fmt.Errorf(string(body))
+		return errors.Errorf(string(body))
```
Comment (cpanato) [inline server/alertmanager/silences.go:55]: github.com/pkg/errors is deprecated, please keep using `fmt.errorf`

File: server/configuration.go [hunk 5]
```diff
@@ -26,8 +30,24 @@ type alertConfig struct {
 	AlertManagerURL string
 }
 
-type configuration struct {
-	AlertConfigs map[string]alertConfig
+func (ac *alertConfig) IsValid() error {
+	if ac.Team == "" {
+		return errors.New("must set a Team")
```
Comment (cpanato) [inline server/configuration.go:35]: since you are adding new things, please move the `"github.com/pkg/errors"` to `errors`

**Review by cpanato — CHANGES_REQUESTED** [review]
lgtm
just a few nits

**Review by cpanato — APPROVED** [review]
thank you

**Pull request merged**